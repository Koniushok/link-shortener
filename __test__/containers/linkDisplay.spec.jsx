import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import LinkDisplay from '../../src/containers/linkDisplay/linkDisplay';

describe('LinkDisplay container', () => {
  const link = {
    id: 'testID',
    title: 'testTitle',
    url: 'testUrl',
    shortLink: 'tUrl',
    clicks: 1,
    tags: ['test', 'test2'],
    description: 'myTest',
  };
  const fetchLink = jest.fn();
  const props = {
    error: false,
    handelClose: () => {},
    linkId: 'testId',
    link,
    loading: false,
    fetchLink,
  };

  describe('LinkDisplay container initial', () => {
    const linkDisplay = shallow(<LinkDisplay {...props} />);
    it('render correctly container', () => {
      expect(shallowToJson(linkDisplay)).toMatchSnapshot();
    });
    it('Call fetchLink method', () => {
      expect(fetchLink).toHaveBeenCalledTimes(1);
    });
  });
  describe('LinkDisplay isLoading', () => {
    const nextProps = {
      ...props,
      loading: true,
    };
    it('render <Loader />', () => {
      const linkDisplay = shallow(<LinkDisplay {...nextProps} />);
      expect(linkDisplay.find('Loader')).toHaveLength(1);
    });
    it('not render <InfoLink />', () => {
      const linkDisplay = shallow(<LinkDisplay {...nextProps} />);
      expect(linkDisplay.find('InfoLink')).toHaveLength(0);
    });
  });
  describe('LinkDisplay isError', () => {
    it('call handelClose', () => {
      const handelClose = jest.fn();
      const nextProps = {
        ...props,
        handelClose,
      };
      const linkDisplay = shallow(<LinkDisplay {...nextProps} />);
      linkDisplay.setProps({ error: true });
      expect(handelClose).toHaveBeenCalledTimes(1);
    });
  });
  it('not call handelClose if LinkDisplay is not error', () => {
    const handelClose = jest.fn();
    const nextProps = {
      ...props,
      handelClose,
    };
    const linkDisplay = shallow(<LinkDisplay {...nextProps} />);
    linkDisplay.setProps();
    expect(handelClose).toHaveBeenCalledTimes(0);
  });
});
