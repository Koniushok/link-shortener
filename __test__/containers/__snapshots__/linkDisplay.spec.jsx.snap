// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkDisplay container LinkDisplay container initial render correctly container 1`] = `
<Modal
  handelClose={[Function]}
  loading={false}
>
  <styled.section>
    <InfoLink
      link={
        Object {
          "clicks": 1,
          "description": "myTest",
          "id": "testID",
          "shortLink": "tUrl",
          "tags": Array [
            "test",
            "test2",
          ],
          "title": "testTitle",
          "url": "testUrl",
        }
      }
    />
  </styled.section>
</Modal>
`;
